node/tests/AuthTest.test.ts

```diff
@@ -0,0 +1,339 @@
+        cmeCluster = clusterAddresses
+            ? await ValkeyCluster.initFromExistingCluster(
+                  true,
+                  parseEndpoints(clusterAddresses),
+                  getServerVersion,
+              )
+            : await ValkeyCluster.createCluster(true, 3, 1, getServerVersion);
+        managementClient = standaloneAddresses
+            ? await GlideClient.createClient({
+                  addresses: cmdCluster
+                      .getAddresses()
+                      .map((addr) => ({ host: addr[0], port: addr[1] })),
+              })
+            : await GlideClusterClient.createClient({
+                  addresses: cmeCluster
+                      .getAddresses()
+                      .map((addr) => ({ host: addr[0], port: addr[1] })),
+              });
+    }, 40000);
+
+    afterEach(async () => {
+        if (managementClient) {
+            try {
+                await managementClient.customCommand(["AUTH", "new_password"]);
+                await managementClient.configSet({ requirepass: "" });
+                await managementClient.flushall();
+            } catch {
+                // Ignore errors
+            }
+
+            try {
+                await client.updateConnectionPassword("");
+            } catch {
+                // Ignore errors
+            }
+        }
+
+        if (cmdCluster) {
+            await flushAndCloseClient(false, cmdCluster.getAddresses());
+        }
+
+        if (cmeCluster) {
+            await flushAndCloseClient(true, cmeCluster.getAddresses());
+        }
+    });
+
+    afterAll(async () => {
+        if (managementClient) {
+            managementClient.close();
+        }
+
+        if (cmdCluster) {
+            await cmdCluster.close();
+        }
+
+        if (cmeCluster) {
+            await cmeCluster.close();
+        }
+    });
+
+    const runTest = async (
+        test: (client: BaseClient) => Promise<void>,
+        protocol: ProtocolVersion,
+        configOverrides?: Partial<BaseClientConfiguration>,
+    ) => {
+        const cluster =
+            configOverrides?.addresses?.length === 1 ? cmdCluster : cmeCluster;
+        const clusterMode = configOverrides?.addresses?.length !== 1;
+
+        const addresses = cluster
+            .getAddresses()
+            .map((addr) =>
+                Array.isArray(addr) ? { host: addr[0], port: addr[1] } : addr,
+            );
+        const options = {
+            addresses,
+            protocol,
+            ...configOverrides,
+        };
+
+        client = clusterMode
+            ? await GlideClusterClient.createClient(options)
+            : await GlideClient.createClient(options);
+
+        try {
+            await test(client);
+        } finally {
+            client.close();
+        }
+    };
+
+    describe.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "update_connection_password_%p",
+        (protocol) => {
+            const NEW_PASSWORD = "new_password";
```

Comment:
I would recommend using a random password for each test case - to make sure that tests don't overlap/interfere

Reply:
I'm afraid I'm using the same one for the same reason, but opposite, you want me not to.
I want to be able, even if a test fails, to recover the server for the whole other tests.
Otherwise, if my tests are failing, it will fail all tests until recreation of the server.